ddtrace/contrib/aiohttp/patch.py

```diff
@@ -64,6 +64,19 @@ async def _create_connection(self, req, *args, **kwargs):
             return result
 
 
+def extract_info_from_url(url):
```

Comment:
This was a rabbit hole of `urlparse` behaviour. Parsing "twice" is the best approach I could come up with, the second parse forces the URL to be absolute so that `netloc` is in the correct position and there's no ambiguity on where the rest of the URL goes in the urlparse 6-tuple - otherwise things go to `scheme` or `path` equivalently with no way of determining which of the 2 is the relevant item.